File: eCommerce/eCommerce.Services/Database/UserActivity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public enum Status{
        Assigned,
        InProgress,
        Completed,
        Cancelled

    }
    public class UserActivity
    {
        public int Id { get; set; }
        public int? UserId { get; set;}
        public User? User { get; set; }
        public int? ActivityId { get; set; }
        public Activity? Activity { get; set; }
        public DateTime? DateAssigned { get; set; }
        public string? Note { get; set;}
        public DateTime? CompletedAt { get; set; }
        public string? RewardTitle { get; set; }
        public DateTime? RewardedAt { get; set; }
        public Status? Status { get; set; }
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/userActivity.dart
import 'package:ecommerce_mobile/model/activity.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';
import 'asset.dart';

part 'userActivity.g.dart';
@JsonSerializable()
class UserActivity {

int? id;
int? userId;
User? user;
int? activityId;
Activity? activity;
DateTime? dateAssigned;
 String? note;
 DateTime? completedAt;
 String? rewardTitle;
 DateTime? rewardedAt;
 String? status;




  UserActivity({
    this.id,this.userId,this.user,this.activityId,this.activity,this.dateAssigned,this.note,this.completedAt,this.rewardTitle,this.rewardedAt,this.status
    
  });

  factory UserActivity.fromJson(Map<String, dynamic> json) => _$UserActivityFromJson(json);

  Map<String, dynamic> toJson() => _$UserActivityToJson(this);


}
